scripts/setup-ssh-keys.sh

```
@@ -0,0 +1,767 @@
+    
+    if [ -f "$failure_file" ]; then
+        failure_count=$(wc -l < "$failure_file")
+        while IFS= read -r host_entry; do
+            failed_hosts+=("$host_entry")
+        done < "$failure_file"
+    fi
+    
+    echo
+    print_success "Successfully distributed keys to $success_count host(s)"
+    
+    if [ $failure_count -gt 0 ]; then
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)